plugins/flytekit-vscode/flytekitplugins/vscode/decorator.py

```diff
@@ -92,45 +146,68 @@ def download_vscode(
 
     # Download remote file to local
     code_server_tar_path = download_file(code_server_remote_path, DOWNLOAD_DIR)
+    
+    plugin_paths = []
+    for plugin in plugins_remote_paths:
+        match = re.search(EXTRACT_EXTENSION_NAME_REGEX, plugin)
+        # Extract the extension_name if a match is found, otherwise skip this plugin
+        if match:
+            extension_name = match.group(1)
+            logger.info(f"Extension Name from plugin url {plugin} : {extension_name}")
+        else:
+            logger.info(f"No match extention name found for path: {plugin}")
+            continue
+        extention_name = re.search(EXTRACT_EXTENSION_NAME_REGEX, plugin).group(1)
+        logger.info("Extention name extracted from the path is {extention_name}")
+        file_path = download_file(plugin, DOWNLOAD_DIR, f"{extention_name}.vsix")
+        plugin_paths.append(file_path)
 
     # Extract the tarball
     with tarfile.open(code_server_tar_path, "r:gz") as tar:
         tar.extractall(path=DOWNLOAD_DIR)
 
-    code_server_dir_path = os.path.join(DOWNLOAD_DIR, code_server_dir_name)
-
-    code_server_bin_dir = os.path.join(code_server_dir_path, "bin")
+    code_server_bin_dir = os.path.join(DOWNLOAD_DIR, code_server_dir_name, "bin")
 
     # Add the directory of code-server binary to $PATH
     os.environ["PATH"] = code_server_bin_dir + os.pathsep + os.environ["PATH"]
+    
+    for p in plugin_paths:
+        logger.info(f"Execute plugin installation command to install plugin {p}")
+        execute_command(f"code-server --install-extension {p}")
 
 
 def vscode(
     _task_function: Optional[Callable] = None,
-    server_up_seconds: Optional[int] = DEFAULT_UP_SECONDS,
+    max_idle_seconds: Optional[int] = MAX_IDLE_SECONDS,
     port: Optional[int] = 8080,
     enable: Optional[bool] = True,
     code_server_remote_path: Optional[str] = DEFAULT_CODE_SERVER_REMOTE_PATH,
```

Comment:
Im considering to wrap all the url related arguments to a dataclass to provide a cleaner interface since users usually won't change the default url.

Can we do

```python
@dataclass
class FlyinConfig:
  code_server_remote_path: DEFAULT_CODE_SERVER_REMOTE_PATH
  code_server_dir_name: DEFAULT_CODE_SERVER_DIR_NAME
  plugins_remote_paths: [...default values]
``` 

And the interface becomes

```python
def vscode(
  config=FlyinConfig(...)
)
```

Note: I will later rename vscode plugin to flyin plugin. So I call the config `FlyinConfig`

Reply:
Done thanks